core/src/main/scala/cats/data/Const.scala

```diff
@@ -179,6 +178,20 @@ sealed abstract private[data] class ConstInstances4 {
     new ConstContravariant[C] {}
 }
 
+trait ConstInstancesBinCompat0 {
+  implicit def catsDataDecideableForConst[D: Monoid]: Decideable[Const[D, ?]] =
```

Comment:
I'm not super sure about this instance, here `product` and `sum` do the same thing, but IMO if we had `Semiring`s, it should be `*` and `+` respectively. Maybe we should remove this instance until we do?

Reply:
Yeah, you're right. Absolutely true, and changing `zero` as discussed above makes this issue more accute.

Reply:
By the way, do we think we'll have `Semiring` constraints available in the near future? Else I'll file an issue for it if we merge this.